ops/testing.py

```diff
@@ -2216,17 +2325,34 @@ def _storage_remove(self, storage_id: str):
         index = int(index)
         self._storage_list[name].pop(index, None)
 
-    def action_get(self):  # type:ignore
-        raise NotImplementedError(self.action_get)  # type:ignore
-
-    def action_set(self, results):  # type:ignore
-        raise NotImplementedError(self.action_set)  # type:ignore
-
-    def action_log(self, message):  # type:ignore
-        raise NotImplementedError(self.action_log)  # type:ignore
-
-    def action_fail(self, message=''):  # type:ignore
-        raise NotImplementedError(self.action_fail)  # type:ignore
+    def action_get(self) -> Dict[str, Any]:
+        action_name = os.environ["JUJU_ACTION_NAME"]
+        params: Dict[str, Any] = {}
+        for name, action_meta in self._meta.actions[action_name].parameters.items():
+            if "default" in action_meta:
+                params[name] = action_meta["default"]
```

Comment:
The long way felt more readable, but this could be a comprehension if preferred:

```suggestion
        params: Dict[str, Any] = {
            name: action_meta["default"]
            for name, action_meta in self._meta.actions[action_name].parameters.items()
            if "default" in action_meta}
```